skt/executable.py

```diff
@@ -194,52 +202,61 @@ def cmd_merge(cfg):
                                         'merge.report')
 
     try:
-        if cfg.get('merge_ref'):
-            report_string = '{}\n{}'.format(
-                report_string,
-                'We merged the following references into the tree:'
-            )
-            for mb in cfg.get('merge_ref'):
-                save_state(cfg, {'mergerepo_%02d' % idx: mb[0],
-                                 'mergehead_%02d' % idx: bhead})
-                (retcode, bhead) = ktree.merge_git_ref(*mb)
-
-                utypes.append("[git]")
-                if retcode:
-                    if retcode != SKT_ERROR:
-                        report_results(merge_result_path, 'false',
-                                       merge_report_path, report_string)
-
-                    return
-
-                report_string += '  - %s\n    into commit %s' % (mb[0],
-                                                                 bhead[:12])
-                idx += 1
-
-        elif cfg.get('patch') or cfg.get('pw'):
-            report_string = 'We applied the following patch(es):\n'
-
-            if cfg.get('patch'):
-                utypes.append("[local patch]")
-                for patch in cfg.get('patch'):
-                    patch = os.path.abspath(patch)
-                    save_state(cfg, {'localpatch_%02d' % idx: patch})
-                    report_string += '  - %s\n' % patch
-
-                    ktree.merge_patch_file(patch)
-                    idx += 1
-
-            elif cfg.get('pw'):
-                utypes.append("[patchwork]")
-                for patch in cfg.get('pw'):
-                    save_state(cfg, {'patchwork_%02d' % idx: patch})
-                    report_string += '  - %s,\n' % get_patch_name(
-                        get_patch_mbox(patch)
-                    )
-                    report_string += '    grabbed from %s\n' % patch
-
-                    ktree.merge_patchwork_patch(patch)
-                    idx += 1
+        if cfg.get('order_merge'):
+            if cfg['order_merge'][0] == 'merge_ref':
```

Comment:
Based on your explanation, this seems wrong. Wouldn't `cfg['order_merge'][0]` be `('option name', 'option value')` so the condition would never go through? And what if we first want to apply a patch and then merge the git repo and the `merge_ref` won't be the first one?